hermeto/core/package_managers/maven/main.py

```diff
@@ -361,6 +396,41 @@ def _get_maven_dependencies(
         else:
             log.warning(f"Missing checksum for {url}, integrity check skipped.")
 
+    # WORKAROUND: Verify .pom file checksums if available
+    # TODO: Remove this workaround once lockfile includes .pom checksums
+    for url, dep_info in deps_to_download.items():
+        if dep_info["checksum_algorithm"]:
+            python_algorithm = _convert_java_checksum_algorithm_to_python(dep_info["checksum_algorithm"])
+            checksum_extension = python_algorithm
+            
+            # Derive .pom paths
+            parsed_url = urlparse(url)
+            path_parts = Path(parsed_url.path)
+            pom_filename = path_parts.stem + ".pom"
+            artifact_dir = download_paths[url].path.parent
+            pom_local_path = artifact_dir / pom_filename
+            pom_checksum_local_path = artifact_dir / f"{pom_filename}.{checksum_extension}"
+            
+            # Verify .pom checksum if checksum file was downloaded
+            if pom_checksum_local_path.exists():
+                try:
+                    with pom_checksum_local_path.open("r") as f:
+                        pom_checksum = f.read().strip().split()[0]  # Take first part in case of additional info
+                    
+                    if pom_local_path.exists():
+                        checksum_info = ChecksumInfo(
+                            algorithm=python_algorithm,
+                            hexdigest=pom_checksum,
+                        )
+                        must_match_any_checksum(pom_local_path, [checksum_info])
+                        log.debug(f"Verified checksum for {pom_local_path}")
+                    else:
+                        log.warning(f"POM file {pom_local_path} not found for checksum verification")
+                except Exception as e:
```

Comment:
This is extremely broad, could you please either make sure it is narrowed or there is a concise explanation why the broadest possible catch is necessary?